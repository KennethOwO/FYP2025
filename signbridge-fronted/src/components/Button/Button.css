:root {
    --primary: #fff;
}

.btn {
    padding: 8px 20px;
    border-radius: 2px;
    outline: none;
    cursor: pointer;
}

.btn-primary {
    background-color: var(--primary);
    color: #242424;
    border: 1px solid var(--primary);
}

.btn--outline {
    background-color: transparent;
    color: #fff;
    padding: 8px 20px;
    border: 1px solid var(--primary);
    transition: all 0.3s ease-out;
}

.btn--outline:hover {
    background-color: var(--primary);
    color: #242424;
    transition: all 0.3s ease-out;
}

.btn--reset,
.btn--submit,
.btn--send,
.btn--accept,
.btn--cancel {
    display: inline-block;
    padding: 10px 20px;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    font-weight: 450;
    text-align: center;
    text-decoration: none;
    box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);
    color: #000;
}

.btn--reset {
    background-color: #ff4d4d;
}

.btn--accept:hover,
.btn--send:hover,
.btn--cancel:hover,
.btn--submit:hover,
.btn--reset:hover {
    color: #fff !important;
    box-shadow: 2px 5px 5px rgba(0, 0, 0, 0.2) !important;
}

.btn--send {
    background-color: #4d85ff !important;
}

.btn--accept {
    background-color: #2ecc71 !important;
}

.btn--cancel {
    background-color: #ff4d4d !important;
}

.btn--submit {
    background-color: #2ecc71 !important;
}

.btn- .btn--medium {
    padding: 8px 20px;
    font-size: 20px;
}

.btn--large {
    padding: 12px 26px;
    font-size: 20px;
}

.btn--medium:hover,
.btn--large:hover {
    color: #242424;
    transition: all 0.3s ease-out;
}
