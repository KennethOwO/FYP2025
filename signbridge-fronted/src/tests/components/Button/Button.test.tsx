//Import necessary libraries -> either "@testing-library/react" or "react-test-renderer"
import { render, fireEvent, screen } from "@testing-library/react";
import { create } from "react-test-renderer";
import { Button } from "../../../components/Button/Button";

//1. Start with describing it in it() function
describe("Test Button Component", () => {
  //2. Write the purposes in the it() function
  it("should have been clicked 1 time", () => {
    //3. Define the necessary props
    const handleClick = jest.fn(); //mock function
    //4. Use the render function from "@testing-library/react"
    render(
      <Button
        type="button"
        onClick={handleClick}
        buttonStyle="btn--primary"
        buttonSize="btn--medium"
      >
        Click Me
      </Button>
    );
    const buttonElement = screen.getByText("Click Me");
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  //This is snapshot testing -> Recommended to do for all components
  it("renders correctly", () => {
    const handleClick = jest.fn();
    //props can be defined like this
    const props = {
      type: "button",
      onClick: handleClick,
      buttonStyle: "btn--outline",
      buttonSize: "btn--large",
    };
    const tree = create(<Button {...props}>Click Me</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
