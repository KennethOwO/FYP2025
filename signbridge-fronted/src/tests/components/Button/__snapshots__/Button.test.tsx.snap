// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component renders correctly 1`] = `
<button
  className="btn btn--outline btn--large"
  data-testid="button-btn--outline"
  onClick={[MockFunction]}
  type="button"
>
  Click Me
</button>
`;
